src/include/migraphx/split_reduce.hpp

```diff
@@ -0,0 +1,45 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+#ifndef MIGRAPHX_GUARD_MIGRAPHX_SPLIT_REDUCE_HPP
+#define MIGRAPHX_GUARD_MIGRAPHX_SPLIT_REDUCE_HPP
+
+#include <migraphx/config.hpp>
+#include <string>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+struct module_pass_manager;
+
+struct MIGRAPHX_EXPORT split_reduce
+{
+    std::size_t split_size = 8192;
```

Comment:
How did we determine 8192 as the size here? Is this some number we've hit somewhere? I understand this is just 2^13 but is there a reason for this?

Reply:
This is just a number I picked after running some different sizes. I maybe adjusting this number further in the future.

Reply:
Do you think its worth tying this to some sort of environment var so we have control of this outside of compiling it in? Seems like something we'd want to test for/tune if it'll make a difference between larger/small models unless the intent was "After a certain point in size we should be split_reducing things" 

More so I'm curious if regardless of model size, is there an optimal split_size for each we can set and see a benifit.

Reply:
I dont think this will need to be a tuning parameter, and I dont see a reason to expose a env variable for this setting. We can alway add one later if we want to do tweak it. This is similiar to the large K value we use for fusing gemms with mlir/ck. We dont have an env variable for it as we haven't needed to tweak it like that.